src/main/java/frc/robot/subsystems/wrist/WristSubsystem.java

```diff
@@ -1,73 +1,221 @@
+import edu.wpi.first.wpilibj.DigitalInput;
+import com.playingwithfusion.TimeOfFlight;
+import com.playingwithfusion.TimeOfFlight.RangingMode;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.WristConstants;
+import lib.factories.SparkMaxFactory;
+import org.littletonrobotics.junction.AutoLog;
 import org.littletonrobotics.junction.Logger;
 
-import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
-import edu.wpi.first.wpilibj2.command.CommandBase;
-import edu.wpi.first.wpilibj2.command.SubsystemBase;
+public class WristSubsystem extends SubsystemBase {
+    private final CANSparkMax m_wristMotor;
+    private final CANSparkMax m_intakeMotor;
+    private final DigitalInput m_wristZeroLimit;
+    private final CANCoder m_wristEncoder;
+    private final PIDController m_wristPID;
+    private final TimeOfFlight m_tofSensor;
+    private final WristIOInputsAutoLogged m_input;
+    // Logging variables
+    private double prevSetpointRaw;
+    private double prevSetpointClamped;
+    private double prevSetpointPID;
+
+    //Shuffleboard data
+    private ShuffleboardTab wristAngleTab;
+    private GenericEntry wristAtSetpointEntry;
+    private GenericEntry wristMotorInvertedEntry;
+    private GenericEntry pieceInsideEntry;
+    private GenericEntry wristAtUpperLimit;
+    private GenericEntry wristAtLowerLimitEntry;
+    private GenericEntry wristEncoderLowerThanLimitEntry;
+    private GenericEntry wristAngleRawEntry;
+    private GenericEntry wristAngleConvertedEntry;
+    private GenericEntry wristTargetEntry;
+    private GenericEntry wristSetpointClampedEntry;
+    private GenericEntry wristPIDOutputEntry;
+    private GenericEntry wristTOFSensorDistanceEntry;
+
+    @AutoLog
+    public static class WristIOInputs {
+        public double wristAngle = 0.0;
+        public double intakeAmps = 0.0;
+    }
 
-public class WristSubsystem extends SubsystemBase{
-    private WristIO m_io;
-    private WristIOInputsAutoLogged m_input;
+    public WristSubsystem() {
 
+        m_input = new WristIOInputsAutoLogged();
 
-    public WristSubsystem(WristIO io){
-        m_io = io;
+        SparkMaxFactory.SparkMaxConfig config = new SparkMaxFactory.SparkMaxConfig();
 
-        m_input = new WristIOInputsAutoLogged();
+        m_wristMotor  = SparkMaxFactory.Companion.createSparkMax(WristConstants.WRIST_ID, config);
+        m_intakeMotor  = SparkMaxFactory.Companion.createSparkMax(WristConstants.INTAKE_ID, config);
+
+        m_wristEncoder = new CANCoder(WristConstants.WRIST_ANGLE_PORT);
+        m_wristEncoder.configSensorInitializationStrategy(SensorInitializationStrategy.BootToZero);
+
+        m_wristZeroLimit = new DigitalInput(WristConstants.LIMIT_SWITCH_PORT);
+
+        m_tofSensor = new TimeOfFlight(WristConstants.TOF_PORT);
```

Comment:
Look into reliability of the sensor data. If the sensor is reading the intake itself, the FOV can be adjusted